src/harness/testcases/WINNF_FT_S_GPR_testcase.py

```diff
@@ -0,0 +1,458 @@
+                                  sas_test_harness_1_config],
+        'domainProxyConfigs': [{
+             'cert': os.path.join('certs', 'dp_1_client.cert'),
+             'key': os.path.join('certs', 'dp_1_client.key')
+        }, {
+             'cert': os.path.join('certs', 'dp_2_client.cert'),
+             'key': os.path.join('certs', 'dp_2_client.key')}],
+        'deltaIap': 2
+    }
+    writeConfig(filename, config)
+
+  @configurable_testcase(generate_GPR_2_default_config)
+  def test_WINNF_FT_S_GPR_2(self, config_filename):
+    """Multiple SAS GWPZ Protection 
+    """
+    config = loadConfig(config_filename)
+    # TODO
+    # test_type= enum (MCP, XPR)
+    # Invoke MCP test steps 1 through 22.
+    # self.executeMcpTestSteps(config, test_type)
+
+  def generate_GPR_3_default_config(self, filename):
+    """ Generates the WinnForum configuration for GPR.3 """
+
+    # Load GWPZ Record
+    gwpz_record_1 = json.load(
+      open(os.path.join('testcases', 'testdata', 'gwpz_record_0.json')))
+
+    # Load devices info
+    device_1 = json.load(
+      open(os.path.join('testcases', 'testdata', 'device_a.json')))
+    # Moving device_1 to a location inside GWPZ zone
+    device_1['installationParam']['latitude'], \
+    device_1['installationParam']['longitude'] = getRandomLatLongInPolygon(gwpz_record_1)
+
+    device_2 = json.load(
+      open(os.path.join('testcases', 'testdata', 'device_b.json')))
+    # Moving device_2 to a location inside GWPZ zone
+    device_2['installationParam']['latitude'], \
+    device_2['installationParam']['longitude'] = getRandomLatLongInPolygon(gwpz_record_1)
+
+    # Load Grant requests
+    grant_request_1 = json.load(
+      open(os.path.join('testcases', 'testdata', 'grant_0.json')))
+    grant_request_1['operationParam']['operationFrequencyRange']['lowFrequency'] = 3650000000
+    grant_request_1['operationParam']['operationFrequencyRange']['highFrequency'] = 3655000000
+
+    grant_request_2 = json.load(
+      open(os.path.join('testcases', 'testdata', 'grant_1.json')))
+    grant_request_2['operationParam']['operationFrequencyRange']['lowFrequency'] = 3660000000
+    grant_request_2['operationParam']['operationFrequencyRange']['highFrequency'] = 3665000000
+
+    # device_b is Category B
+    # Load Conditional Data
+    self.assertEqual(device_2['cbsdCategory'], 'B')
+    conditionals_device_2 = {
+        'cbsdCategory': device_2['cbsdCategory'],
+        'fccId': device_2['fccId'],
+        'cbsdSerialNumber': device_2['cbsdSerialNumber'],
+        'airInterface': device_2['airInterface'],
+        'installationParam': device_2['installationParam'],
+        'measCapability': device_2['measCapability']
+    }
+
+    conditionals = [conditionals_device_2]
+
+    # Remove conditionals from registration
+    del device_2['cbsdCategory']
+    del device_2['airInterface']
+    del device_2['installationParam']
+    del device_2['measCapability']
+
+    # Load GWPZ Record
```

Comment:
Please remove, since we already loaded the GWPZ record in Lines 368-370.

Reply:
Removed. Thank you.